tests/hikari/test_guilds.py

```diff
@@ -753,7 +753,85 @@ def test_make_banner_url_when_hash(self, model):
         model.banner_hash = None
         assert model.make_banner_url(ext="png", size=2048) is None
+    
+    @pytest.mark.asyncio
+    async def test_fetch_owner(self, model):
+        model.app.rest.fetch_member = mock.AsyncMock()
+
+        owner = await model.fetch_owner()
+
+        model.app.rest.fetch_member.assert_awaited_once_with(123, 1111)
+        assert owner is model.app.rest.fetch_member.return_value
+    
+    @pytest.mark.asyncio
+    async def test_fetch_widget_channel(self, model):
+        model.app.rest.fetch_channel = mock.AsyncMock()
+
+        widget_channel = await model.fetch_widget_channel()
+
+        assert widget_channel is model.app.rest.fetch_channel.return_value
+        model.app.rest.fetch_channel.assert_awaited_once_with(192729)
+    
+        model.widget_channel_id = None
+
+        widget_none_case = await model.fetch_widget_channel()
+        assert None is widget_none_case
+    
+    @pytest.mark.asyncio
+    async def test_fetch_rules_channel(self, model):
+        model.app.rest.fetch_channel = mock.AsyncMock()
+
+        rules_channel = await model.fetch_rules_channel()
+
+        assert rules_channel is model.app.rest.fetch_channel.return_value
+        model.app.rest.fetch_channel.assert_awaited_once_with(123445)
+
+        model.rules_channel_id = None
+
+        rules_none_case = await model.fetch_rules_channel()
+        assert None is rules_none_case
+
+    @pytest.mark.asyncio
+    async def test_fetch_system_channel(self, model):
+        model.app.rest.fetch_channel = mock.AsyncMock()
+
+        system_channel = await model.fetch_system_channel()
+
+        assert system_channel is model.app.rest.fetch_channel.return_value
+        model.app.rest.fetch_channel.assert_awaited_once_with(123888)
+
+        model.system_channel_id = None
+
+        system_none_case = await model.fetch_system_channel()
+        assert None is system_none_case
+
+    @pytest.mark.asyncio
+    async def test_fetch_public_updates_channel(self, model):
+        model.app.rest.fetch_channel = mock.AsyncMock()
+
+        public_updates_channel = await model.fetch_public_updates_channel()
+
+        assert public_updates_channel is model.app.rest.fetch_channel.return_value
+        model.app.rest.fetch_channel.assert_awaited_once_with(99699)
+
+        model.public_updates_channel_id = None
+
+        public_updates_none_case = await model.fetch_public_updates_channel()
+        assert None is public_updates_none_case
+
+    @pytest.mark.asyncio
+    async def test_fetch_afk_channel(self, model):
+        model.app.rest.fetch_channel = mock.AsyncMock()
+
+        afk_channel = await model.fetch_afk_channel()
+
+        assert afk_channel is model.app.rest.fetch_channel.return_value
+        model.app.rest.fetch_channel.assert_awaited_once_with(1234)
+
+        model.afk_channel_id = None
 
+        afk_none_case = await model.fetch_afk_channel()
+        assert None is afk_none_case
```

Comment:
This should be a different test case all together

Reply:
Are you suggesting to combine these type of tests into one function?

Reply:
No, have 2 tests. 1 for `fetch_afk_channel` when None and one when its set:

```py
async def fetch_afk_channel():
    ...

async def fetch_afk_channel_when_None():
    ...
```

You can have a look at the existing tests for reference

Reply:
Ah, will change these too